### PR TITLE
dmllib.h: Only use `VT_log_warning` if available

### PR DESCRIPTION
**Pull request by lwaern-intel**
Otherwise, use `SIM_LOG_ERROR`. Needed for compatibility with packages with a Base dependency lower than 6.0.188.

Gonna run PackageTest on this to make sure it works.

### CODE REVIEW

**Comment by lwaern-intel** [comment]
[PackageTest L run](https://teamcity01-ir.devtools.intel.com/viewQueued.html?itemId=13479430)

**Comment by syssimics** [comment]
Verification [#13479403](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13479403): ✅ pass

File: include/simics/dmllib.h [hunk 1]
```diff
@@ -41,6 +41,22 @@ static inline uint64 DML_shlu(uint64 a, uint64 b)
...
 
+// This is used instead of SIM_LOG_WARNING directly, as a SIM_get_api_function
+// indirection is needed to work with old 6 base packages
+// TODO scrap this once the base dependency of every package is bumped
+// to 6.0.188 or above.
+#define _DMLLIB_LOG_WARNING(dev, grps, ...) do {                              \
+    void (*log_warning)(conf_object_t *NOTNULL, uint64, const char *NOTNULL,  \
+                        ...)                                                  \
+        = (void (*)(conf_object_t *NOTNULL, uint64, const char *NOTNULL, ...) \
+          )SIM_get_api_function("VT_log_warning");                            \
```
Comment (jhbaarnh) [inline include/simics/dmllib.h:52]: Does the `GET_API_FUNCTION` macro help?

Comment (lwaern-intel) [reply include/simics/dmllib.h:52]: Oh, interesting, I was not aware of it. It sure would.

**Comment by syssimics** [comment]
Verification [#13479845](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13479845): ✅ pass

**Comment by lwaern-intel** [comment]
[Actual functioning PackageTest L run](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=13480604). Not finished yet, but it's already passing tests that previously failed, so yeah, this seems to be doing the trick.

**Review by mandolaerik — APPROVED** [review]

**Pull request merged**